### PR TITLE
Fixed issue #18268: Unable to get any page debug set an set_time_limit disable

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: just silence action and log it as error.

### CODE REVIEW

**Review by gabrieljenik — APPROVED** [review]

**Comment by gabrieljenik** [comment]
Maybe a helper function like safe_ini_set could help?
It could try and log safely.

**Comment by Shnoulle** [comment]
We use only ini_set here. Unsure it's needed

File: application/core/LSYii_Controller.php [hunk 1]
```diff
@@ -103,17 +103,15 @@ protected function customInit()
...
+            if(!@set_time_limit(App()->getConfig('max_execution_time'))) {
+                Yii::log("Unable to set time limit to " . App()->getConfig('max_execution_time'), \CLogger::LEVEL_WARNING, 'application.controller');
+            }
         }
         if (ini_get('memory_limit') != -1 && convertPHPSizeToBytes(ini_get("memory_limit")) < convertPHPSizeToBytes(Yii::app()->getConfig('memory_limit') . 'M')) {
-            try {
-                ini_set("memory_limit", Yii::app()->getConfig('memory_limit') . 'M'); // Set Memory Limit for big surveys
-            } catch (Exception $e) {
-            };
+            if (@ini_set("memory_limit", Yii::app()->getConfig('memory_limit') . 'M') === false) {
```
[the remark this answers is not in the document]
Comment (olleharstedt) [reply application/core/LSYii_Controller.php:112]: Oh that wouldn't check for `false` I guess.

[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/core/LSYii_Controller.php:112]: ?
https://www.php.net/manual/en/function.ini-set.php

>  Returns the old value on success, false on failure. 

I think it's clear enough ?
Why use a try catch when we can just silence the error. With @ : no error, no need try/cacth ....

**Comment by tiborpacalat** [comment]
@Shnoulle how can I test this? Is this still relevant?

**Comment by Shnoulle** [comment]
Oh ?

This need an update for 8.1

**Pull request merged**